pkg/workload/pgx_helpers.go

```diff
@@ -357,8 +378,6 @@ func (m *MultiConnPool) WarmupConns(ctx context.Context, numConns int) error {
 	estConns := make([]*pgxpool.Conn, 0, numWarmupConns)
 	defer func() {
 		for _, conn := range estConns {
-			// NOTE(seanc@): Release() connections before canceling the warmupCtx to
-			// prevent partially established connections from being Acquire()'ed.
 			conn.Release()
 		}
```

Comment:
I can't comment on it but line 395 below (`break WARMUP`) is dead code. When a `Done()` channel is closed, `Err()` is guaranteed to return an error (that's in the contract) so you just need a blanket `return err` inside of the `case`.